test/id.test.js

```diff
@@ -247,3 +246,152 @@ describe('mongodb default id name', function() {
+      found.toObject().name.should.equal('John');
+    });
+
+    it('should find model with ObjectID-like id', async function() {
+      const user = await User.create({id: objectIdLikeString, name: 'John'});
+      user.id.should.be.an.instanceOf(ds.ObjectID);
+      user.id.should.eql(ObjectID(objectIdLikeString));
+      const found = await User.findById(objectIdLikeString);
+      found.toObject().name.should.equal('John');
+    });
+
+    it('should find model with string id', async function() {
+      const user = await User.create({id: 'a', name: 'John'});
+      user.id.should.be.an.instanceOf(String);
+      user.id.should.equal('a');
+      const found = await User.findById('a');
+      found.toObject().name.should.equal('John');
+    });
+  });
+
+  context('set to true', function() {
+    const User = ds.createModel(
+      'user2',
+      {
+        id: {type: String, id: true},
+        name: String,
+      },
+      {strictObjectIDCoercion: true}
+    );
+
+    beforeEach(function(done) {
+      User.deleteAll(done);
+    });
+
+    it('should not coerce to ObjectID', async function() {
+      const user = await User.create({id: objectIdLikeString, name: 'John'});
+      user.id.should.equal(objectIdLikeString);
+    });
+
+    it('should not find model with ObjectID id', async function() {
+      const user = await User.create({name: 'John'});
+      const found = await User.findById(user.id);
+      (found === null).should.be.true();
+    });
+
+    it('should find model with ObjectID-like string id', async function() {
+      const user = await User.create({id: objectIdLikeString, name: 'John'});
+      user.id.should.not.be.an.instanceOf(ds.ObjectID);
+      user.id.should.eql(objectIdLikeString);
+      const found = await User.findById(objectIdLikeString);
+      found.toObject().name.should.equal('John');
+    });
+
+    it('should find model with string id', async function() {
+      const user = await User.create({id: 'a', name: 'John'});
+      user.id.should.be.an.instanceOf(String);
+      user.id.should.equal('a');
+      const found = await User.findById('a');
+      found.toObject().name.should.equal('John');
+    });
+  });
+
+  context('set to true, id type set to ObjectID', function() {
+    const User = ds.createModel(
+      'user3',
+      {
+        id: {type: String, id: true, mongodb: {dataType: 'ObjectID'}},
+        name: String,
+      },
+      {strictObjectIDCoercion: true}
+    );
+
+    beforeEach(function(done) {
+      User.deleteAll(done);
+    });
+
+    it('should coerce to ObjectID', async function() {
+      const user = await User.create({id: objectIdLikeString, name: 'John'});
+      user.id.should.be.an.instanceOf(ds.ObjectID);
+    });
+
+    it('should throw if id is not a ObjectID-like string', async function() {
+      // Why is this not working?
+      // await User.create({id: 'abc', name: 'John'}).should.be.rejectedWith(/not an ObjectID string/);
```

Comment:
Here are two ways of doing it (one is for expect/chai, one for mocha):
expect:
https://github.com/strongloop/loopback-next/blob/75939a4a144b3068ff2890394c178faad58d7458/packages/cli/test/integration/generators/relation.integration.js#L39-L48

Mocha:
https://github.com/strongloop/loopback-datasource-juggler/blob/91ab3e916281e56e8675889ab7c5a31dc223c22c/test/datasource.test.js#L446-L451

I suggest trying to have a promise based test:
```js
it('should throw if id is not an ObjectID-like string', () => {

  return user.create({id: 'abc', name: 'John'}).should.be.rejectedWith(/not an ObjectID string/);
});
```

Reply:
> return user.create({id: 'abc', name: 'John'}).should.be.rejectedWith(/not an ObjectID string/);

Doesn't help.

Would it be ok to replace `should` with `expect` in this repo?

Reply:
Maybe my example is wrong. I didn't get to try it out. I'll try it out and let you know. In terms of replacing `should` with `expect`, it is ok, but not sure how much effort that is and how much beneficial it is.